addons/netfox.extras/window-tiler.gd

```diff
@@ -0,0 +1,61 @@
+extends Node
+
+
+func _ready() -> void:
+	EngineDebugger.register_message_capture("tile-session", _on_tile_session)
+	EngineDebugger.send_message.call_deferred("tile-session:get_id", [])
+
+
+func _on_tile_session(message: String, data: Array) -> bool:
+	if message == "session_id":
+		var info: Dictionary = data[0]
+		var id = info.get("id", -1)
+		var total = info.get("total", 0)
+		if total > 1:
+			for i in range(total):
+				if i == id:
+					tile_window(i, total)
+		return true
+	return false
+
+
+func tile_window(i: int, total: int) -> void:
+
+	if not ProjectSettings.get_setting("netfox/extras/auto_tile_windows", true):
+		return
+
+	var screen = ProjectSettings.get_setting("netfox/extras/screen", 0)
+	var screen_rect = DisplayServer.screen_get_usable_rect(screen)
+
+	var window: Window = get_tree().get_root()
+	window.set_current_screen(screen)
+	window.borderless = ProjectSettings.get_setting("netfox/extras/borderless", true)
+
+	# Divide up the screen
+
+	var windows_per_row = int(ceil(sqrt(total)))
+	var windows_per_col = int(ceil(total / float(windows_per_row)))
+	var window_size = Vector2(
+		screen_rect.size.x / windows_per_row, screen_rect.size.y / windows_per_col
+	)
+
+	# Set up window size, don't take up more than 1/4 of the screen.
```

Comment:
Why this check? With 1 window nothing happens, with 2 windows they will cover the left and right side of the screen, and with more, they will end up as quarters or smaller. 

Is this to avoid the 2 window case?

Reply:
Yeah, originally it tiled top to bottom which I later changed. I'll clean this up.